tests/native_tests/language.hy

```diff
@@ -1592,3 +1592,14 @@
   "Make sure argument destructuring works"
   (defn f [[a b] [c]] (, a b c))
   (assert (= (f [1 2] [3]) (, 1 2 3))))
+
+#@(pytest.mark.xfail
+(defn test-assert-multistatements []
+  ; https://github.com/hylang/hy/issues/1390
+  (setv l (set))
+  (defn f [x]
+    (.add l x)
+    True)
+  (assert (do (f 1) (f 2)) (do (f 3) (f 4)))
+  (print l)
+  (assert (= l [1 2 3 4]))))
```

Comment:
This seems unrelated to the rest of the PR.

Reply:
It is. I put that commit in because I noticed the bug while reading through the compiler. I figured that adding an xfail test wasn't worth its own PR.